posts/2015-09-07-idempotent-encoders-decoders.md

```diff
@@ -5,7 +5,7 @@ tags: codec
 title: Idempotent encoders and decoders
 ---
 
-Encoding and decoding data is a practice that has been present for a long time since we started storing and transmiting information, even before the Information Age. Encoding and decoding with a specific algorithm, or as we are used to call it, a `codec`, has several different benefits such as: reduced storage space, faster transmission over a selected transport, possibility to capture data in a specific system readable form, between others. However not every codec has to offer the same features and it is typically wise to use the best codec for the type of data we are dealing with.
+Encoding and decoding data is a practice that has been present since we started storing and transmiting information -- even before the Information Age. Encoding and decoding with a specific algorithm, or as we are used to call it, a `codec`, has several different benefits such as: reduced storage space, faster transmission over a selected transport, the possibility to capture data in a specific system-readable form, among others. However not every codec has offers the same features and it is typically wise to use the best codec for the type of data we are dealing with.
 
 A deterministic and idempotent encoding and decoding process should return the same data that was served for the encode function, once the decode function is applied. In essence, something like:
```

Comment:
What does deterministic mean in this context?
What does idempotent mean?
You should explain, maybe.